### PR TITLE
Core Lib Documentation:  `Index` module 

### PR DESCRIPTION
**Pull request by TAdev0**

### CODE REVIEW

File: corelib/src/ops/index.cairo [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
 use crate::traits::IndexView as DeprecatedIndexView;
 #[feature("deprecated-index-traits")]
 use crate::traits::Index as DeprecatedIndex;
 
-/// The following two traits are for implementing the [] operator. Only one should be implemented
-/// for each type. Both are not consuming of self, the first gets a snapshot of the object and
-/// the second gets ref.
-/// Trait for a view of an item contained in type `C` with an index of type `I`.
+
+/// A trait for a view of an item contained in type `C` with an index of type `I`.
```
Comment (enitrat) [inline corelib/src/ops/index.cairo:17]: mention the `[ ]` operator at the trait level

**Review by enitrat — CHANGES_REQUESTED** [review]
see https://doc.rust-lang.org/src/core/ops/index.rs.html#58

File: corelib/src/ops/index.cairo (hunk 1 above)
Comment (enitrat) [reply corelib/src/ops/index.cairo:17]: mention that it should not be implemented if Index already exists on that type and vice versa

Comment (TAdev0) [reply corelib/src/ops/index.cairo:16]: addressed @enitrat

Comment (enitrat) [reply corelib/src/ops/index.cairo:16]: this still gives no insights on the reasons I should implement a trait and not the other

Comment (enitrat) [reply corelib/src/ops/index.cairo:16]: and no actionable example :(

**Review by enitrat — APPROVED** [review]
I was expecting something more like this

**Pull request closed without merging**